src/app.py

```diff
@@ -41,7 +41,10 @@ def _verify_config(cfg):
         else:
             st = os.stat(ssh_key)
             if st.st_uid != os.getuid() or (st.st_mode & 0o7777) not in (0o700, 0o600, 0o400):
-                raise PermissionError(ssh_key)
+                if cfg["IGNORE_SECRET_PERMS"]:
+                    app.logger.info("Ignoring permissions/ownership issues on " + ssh_key)
```

Comment:
What's the deal with ignoring the ssh_key perms?  Won't ssh just error out later anyway if they are bad?

Reply:
I've been testing with docker and docker-compose so far (not yet running in k8s). When I bind mount the SSH key into the container, it inherits the uid of the host filesystem. If I set the permissions to 600, Apache can't read the file.

You could probably set the host filesystem owner to match the apache uid. This is a little messy. And it would probably still leave an issue with k8s. The docs for k8s secrets indicate you can set permissions, but not ownership.

Reply:
Ah, i see.  So you are setting it to be world readable  Well whatever works.  I was just worried that if the above permissions check failed, then ssh would also complain.

For instance if i chmod a key file to 644:

```
$ ssh-add ./github_id_dsa
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@         WARNING: UNPROTECTED PRIVATE KEY FILE!          @
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
Permissions 0644 for './github_id_dsa' are too open.
It is required that your private key files are NOT accessible by others.
This private key will be ignored.
```

Reply:
In testing, it cloned the bitbucket repo and displayed contacts OK. Perhaps `ssh -i keyfile` isn't as strict?

Reply:
Ok, sounds like it works then, which is the main thing!